Shared/UserAgent.swift

```diff
@@ -35,33 +35,113 @@ open class UserAgent {
+            return getUserAgent(domain: domain, platform: .Mobile)
         }
     }
 }
+
+public enum UserAgentPlatform {
+    case Desktop
+    case Mobile
+}
+
+public struct CustomUserAgentConstant {
+    public static let mobileUserAgent = [
+        "whatsapp.com":
+        UserAgentBuilder.defaultMobileUserAgent().userAgent(),
+        "paypal.com":
+        UserAgentBuilder.defaultMobileUserAgent().userAgent(),
+        "yahoo.com":
+        UserAgentBuilder.defaultMobileUserAgent().userAgent()]
+    public static let desktopUserAgent = [
+        "whatsapp.com":
+        UserAgentBuilder.defaultDesktopUserAgent().clone(extensions: "Version/\(AppInfo.appVersion) \(UserAgent.uaBitSafari)"),
+        "paypal.com":
+        UserAgentBuilder.defaultDesktopUserAgent().clone(extensions: "Version/\(AppInfo.appVersion) \(UserAgent.uaBitSafari)"),
+        "yahoo.com":
+        UserAgentBuilder.defaultDesktopUserAgent().clone(extensions: "Version/\(AppInfo.appVersion) \(UserAgent.uaBitSafari)")]
+}
+
+public struct UserAgentBuilder {
+    // User agent components
+    fileprivate var product = ""
+    fileprivate var systemInfo = ""
+    fileprivate var platform = ""
+    fileprivate var platformDetails = ""
+    fileprivate var extensions = ""
+    
+    init(product: String, systemInfo: String, platform: String, platformDetails: String, extensions: String) {
+        self.product = product
+        self.systemInfo = systemInfo
+        self.platform = platform
+        self.platformDetails = platformDetails
+        self.extensions = extensions
+    }
+    
+    public func userAgent() -> String {
+        let userAgentItems = [product, systemInfo, platform, platformDetails, extensions]
+        return removeEmptyComponentsAndJoin(uaItems: userAgentItems)
+    }
+    
+    public func clone(product: String? = nil, systemInfo: String? = nil, platform: String? = nil, platformDetails: String? = nil, extensions: String? = nil) -> String {
+        let userAgentItems = [product ?? self.product, systemInfo ?? self.systemInfo, platform ?? self.platform, platformDetails ?? self.platformDetails, extensions ?? self.extensions]
+        return removeEmptyComponentsAndJoin(uaItems: userAgentItems)
+    }
+    
+    /// Helper method to remove the empty components from user agent string that contain only whitespaces or are just empty
+    private func removeEmptyComponentsAndJoin(uaItems: [String]) -> String {
+        return uaItems.filter{ !$0.isEmptyOrWhitespace() }.joined(separator: " ")
+    }
+    
+    public static func defaultMobileUserAgent() -> UserAgentBuilder {
+        return UserAgentBuilder(product: "Mozilla/5.0", systemInfo: "(\(UIDevice.current.model); CPU OS \(UIDevice.current.systemVersion.replacingOccurrences(of: ".", with: "_")) like Mac OS X)", platform: "AppleWebKit/605.1.15 ", platformDetails: "(KHTML, like Gecko)", extensions: "FxiOS/\(AppInfo.appVersion)  \(UserAgent.uaBitMobile) \(UserAgent.uaBitSafari)")
```

Comment:
nit: static constant in this class for "AppleWebKit/605.1.15 " since we have to keep updating that now.

Reply:
Moved AppleWebKit/605.1.15 to the class above and made it a static constant.